### DISCUSSION TITLE
Issues building packages without default config (in /etc)

### DISCUSSION THREAD

**Discussion by SpidFightFR** [void-linux/void-packages#46338] [Q&A] [answered]
Hey everyone, i need help with xbps-src.

I try to make a template for fastfetch (for my personal usage).

This package doesn't come with a default configuration file. But you can generate one using a command (like sddm): `fastfetch --gen-config {/output/file/name.conf}`.

This leads me to an error at the end of compilation:

```
=> fastfetch-2.0.5_1: running pre-pkg hook: 99-pkglint ...
=> ERROR: fastfetch-2.0.5_1: '/etc/fastfetch/config.conf' configuration file not in PKGDESTDIR!
=> ERROR: fastfetch-2.0.5_1: cannot continue with installation!
```

I tried:
- to compile it by default (well it's a normal first try)
- to create a conf file using another linux, to copy it into `srcpkgs/fastfetch/files/config.conf` and use vconf in `pre_check()` (feels like a dirty workaround but why not)

Anyways, i'm running out of ideas... any help is welcome !
If you need anything, please ask.

**Comment by classabbyamp** [comment]
`conf_files` is only for marking a file that exists in a package as specially mutable, i. e., if it is modified by the user after installation, it isn't removed/overwritten and instead a `.new` file is generated with the new content. You don't put something in `conf_files` if it doesn't exist in the package, as xbps can't/won't track it in the first place and thus will never overwrite it.

**Reply by SpidFightFR** [reply]
Why does this error show up, even when the template doesn't mention any config file ? I'm quite confused... what am i missing?

**Reply by Duncaen** [reply, marked as answer]
There has to be, it uses the `$conf_files` which comes from the template:
https://github.com/void-linux/void-packages/blob/7d9464574670939003a26a359594981925e24cc9/common/hooks/pre-pkg/99-pkglint.sh#L73-L78

**Reply by SpidFightFR** [reply]
> There has to be, it uses the `$conf_files` which comes from the template:
> 
> https://github.com/void-linux/void-packages/blob/7d9464574670939003a26a359594981925e24cc9/common/hooks/pre-pkg/99-pkglint.sh#L73-L78

Welp... i must be stupid. 
I just re-written the template from scratch, to be sure to miss nothing, and it worked. 
Perhaps i forgot something before.

My apologies for the disturbance. :sweat: 
Thank you to the both of you for your help tho :heart:
